ui/shared/components/PopupWithModal.jsx

```diff
@@ -0,0 +1,49 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { connect } from 'react-redux'
+import { Popup } from 'semantic-ui-react'
+
+import { getOpenModals } from 'redux/utils/modalReducer'
+
+const BehindModalPopup = styled(Popup)`
+  z-index: 500 !important;
+`
+
+class PopupWithModal extends React.PureComponent {
```

Comment:
Why is a PopupWithModal component needed?

Reply:
because when you click the edit button a modal popped up, but then when you would move your mouse to the modal the popup would close and the modal would unmount

Reply:
can the modal be mounted/rendered by some top-level component outside the popup, and just shown/hidden based on a boolean in redux state?

Reply:
(I'll merge as is now, and if you agree, this can also go in a future PR)

Reply:
technically yes, but not without entirely refactoring how `BaseFieldView` works and the redux state bit would be kind of messy. I'm also not sure why thats any cleaner than this. 

The modal isn't rendering in the DOM inside the popup, btw. Its just the way semantic UI controls it it is still unmounting even though it was mounted on the main document body

Reply:
I'm confused about this.. need to play around with this in my local dev instance.

Reply:
We can also talk in person if that helps